src/test/java/org/apache/sysds/test/functions/io/proto/FrameReaderWriterProtoTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysds.test.functions.io.proto;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Random;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.sysds.common.Types;
+import org.apache.sysds.runtime.io.FrameReader;
+import org.apache.sysds.runtime.io.FrameReaderFactory;
+import org.apache.sysds.runtime.io.FrameWriter;
+import org.apache.sysds.runtime.io.FrameWriterFactory;
+import org.apache.sysds.runtime.matrix.data.FrameBlock;
+import org.apache.sysds.runtime.util.DataConverter;
+import org.apache.sysds.test.TestUtils;
+import org.apache.wink.json4j.JSONException;
+import org.junit.Test;
+
+public class FrameReaderWriterProtoTest
+{
+	private static final String FILENAME_SINGLE = "target/testTemp/functions/data/FrameReaderWriterProtoTest/testFrameBlock.proto";
+
+	@Test
+	public void testWriteReadFrameBlockSingleSingleFromHDFS() throws IOException, JSONException {
+		testWriteReadFrameBlockComplete(1,1, 4669201);
+	}
```

Comment:
Since these tests are the same as in `i0/json/freameReaderWriterJSONLTest`.
You could make it into a class that extends and overwrites the json FrameReaderWriter, or another super class for both of them.

Reply:
Good point, will refactor that.

Reply:
I refactored and simplified the tests:
- removed unnecessary JSONException which is never thrown in this code (so it does not make sense to use inheritance any more).
- simplified method calls

With the current implementation it should be clear what happens in every test.